tests/rust-axum/src/main.rs

```diff
@@ -321,7 +321,8 @@ async fn run_server(database: Arc<Database>, messagebus: Arc<MessageBus>, port:
         .merge(accessories)
         .merge(health)
         .merge(locations)
-        .with_state(state);
+        .with_state(state.clone()) // todo: debug state issue
+        .layer(Extension(state));
```

Comment:
i think that when we use with_state Extension(state) should be omitted, isnt it?

Reply:
yup, it should

I wanted to add support for `with_state` without understanding the concept and it quickly turned out that it's hardcoded in the original `Handler` concept. Because of current implementation I cannot extend it. Axum is not propagating state to generated custom routers.

There are imo two options:
- don't support `with_state` in generated Router (+ make generating of router optional) - if someone really wants to use state, he can use original axum routers directly
- investigate the issue further and probably create MR to axum so we can mitigate same behaviour like in original standard router

That is why this MR is still in progress. BTW IMO `with_state` is a micro optimization of Extension concept - thanks to it you can skip one hashset (extensions) lookup during each call and access one global variable directly :D Extension approach is still acceptable: https://docs.rs/axum/latest/axum/index.html#using-request-extensions

Reply:
ok, relaxed router added
but cant remove with_state in strict router in tests as it fails with error defined in templates `state extension missing. This is a bug in code schema-tools, please file an issue`